main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -1477,11 +1477,11 @@ public static boolean deleteWaypoint(final int id) {
     }
 
     private static void saveSpoilersWithoutTransaction(final Geocache cache) {
-        final String geocode = cache.getGeocode();
-        database.delete(dbTableSpoilers, "geocode = ?", new String[]{geocode});
-
         final List<Image> spoilers = cache.getSpoilers();
         if (CollectionUtils.isNotEmpty(spoilers)) {
+            final String geocode = cache.getGeocode();
+            database.delete(dbTableSpoilers, "geocode = ?", new String[]{geocode});
```

Comment:
I tried with PreparedStatement, but the saveSpoilers method takes only 0,1 % of the Cache store operation time. But having it is not a bad thing, so I keep it in.